plugins/gimp/file-jxl-save.cc

```diff
@@ -789,14 +779,24 @@ bool SaveJpegXlImage(const gint32 image_id, const gint32 drawable_id,
     return false;
   }
 
-  // convert precision and colorspace
-  if (jxl_save_opts.is_linear &&
-      jxl_save_opts.basic_info.bits_per_sample < 32) {
-    gimp_image_convert_precision(duplicate, GIMP_PRECISION_FLOAT_LINEAR);
+  // babl destination format for encoder, used inside loop
+  babl_init();
+  if (jxl_save_opts.icc_attached) {
+    jxl_save_opts.SetModel(jxl_save_opts.is_linear);
   } else {
-    gimp_image_convert_precision(duplicate, GIMP_PRECISION_FLOAT_GAMMA);
+    jxl_save_opts.SetModel(!jxl_save_opts.is_linear);
   }
+  jxl_save_opts.pixel_format.data_type = JXL_TYPE_FLOAT;
+  jxl_save_opts.SetBablType("float");
+  const Babl* destination_format =
+      babl_format(jxl_save_opts.babl_format_str.c_str());
 
+  // convert precision to float if needed
+  if (jxl_save_opts.basic_info.bits_per_sample < 32) {
+    gimp_image_convert_precision(duplicate, GIMP_PRECISION_FLOAT_LINEAR);
+  } else if (!jxl_save_opts.is_linear) {
+    gimp_image_convert_precision(duplicate, GIMP_PRECISION_FLOAT_LINEAR);
```

Comment:
`if (jxl_save_opts.basic_info.bits_per_sample < 32 || !jxl_save_opts.is_linear) {` ?

Reply:
Yes.  Thank you.  Done.